src/lib.rs

```diff
@@ -360,7 +362,12 @@ struct MemberInner {
 }
 
 impl Member {
-    fn new(base_ty: syn::Type, f: syn::Field, offset: usize) -> syn::Result<Self> {
+    fn new(
+        base_ty: syn::Type,
+        f: syn::Field,
+        offset: usize,
+        endianness: Endianness,
```

Comment:
The extra endianness parameter is not strictly necessary. We could calculate the offset in the caller instead, depending on the bit order (see next comment for naming discussion).

```rust
// FIXME: offset > bits leads to panic!
let o = if order == Order::Msb { bits - offset } else { offset };
let f = Member::new(ty.clone(), field, o)?;
```
> https://github.com/wrenger/bitfield-struct-rs/pull/19/commits/5281d4b4b0f2c86c119b3561fd49eb9cfeada8ef#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759L257

Reply:
having a problem with this implementation... the field offset calculation depends on the number of bits in the field itself, which isn't known until `Member::new(...)` is called, right?

the MSB test fails on the first `.with_short(...)` call, since `offset` is calculated as 32 and it's not happy about shifting left 32 bits. should be `bits - offset - field_bits`, so I think we do have to pass in `order`.

I also realized that in `Member::new(...)`, I was only using `Self::#offset_ident` in the calculations because that's how I was getting the total bits from `base_ty`. that could also be passed in to keep using the `#offset` literal in the calculations.

Reply:
Ah yes, you are correct, we have to also subtract the size of the member itself.
So we have to do it in `Member::new` after `parse_field`, where we get its size in bits.

Then I would suggest passing `order` and `base_bits` into `Member::new` and then computing the offset there. Then we do not have to add extra fields to the `Member` struct and also `to_tockens` can use the literal.

E.g.: in `Member::new`
```rust
// ...
if bits > 0 && !ignore {
    // Compute the offset
    if offset + bits > base_bits {
        return Err(syn::Error::new(ty.span(), "The total size of the members is too large!"));
    }
    let offset = if order == Order::Lsb { offset } else { base_bits - offset - bits };
    // ...
```

Reply:
alright, think I got it all wrapped up. thanks again!